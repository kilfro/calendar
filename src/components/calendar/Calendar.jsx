import '@style/calendar/calendar.less'

import React, { useEffect } from 'react'

import Month from './Month'
import MonthLabel from './MonthLabel'
import Navigation from './Navigation'
import WeekdayLabel from './WeekdayLabel'
import { connect } from 'react-redux'
import { goToToday } from '../../store/actions'

export const Calendar = ({ goToToday }) => {
    useEffect(() => { goToToday() }, [])

    return (
        <div className='calendar'>
            <Navigation />
            <MonthLabel />
            <WeekdayLabel />
            <Month />
        </div>
    )
}

const mapDispatchToProps = {
    goToToday
}

export default connect(null, mapDispatchToProps)(Calendar)