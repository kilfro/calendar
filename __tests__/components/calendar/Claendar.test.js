import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import Calendar from '../../../src/components/calendar/Calendar'

describe('Calendar', () => {
    test('should render correctly', () => {
        const state = {
            now: new Date(),
            selected: new Date(),
            month: new Date(),
            tasksMap: {}
        }
        const store = configureStore([])(state)
        const component = mount(
            <Provider store={store}>
                <Calendar />
            </Provider>
        )

        expect(component).toMatchSnapshot()
    })

})
