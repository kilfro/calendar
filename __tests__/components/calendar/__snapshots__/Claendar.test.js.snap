// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Calendar>
    <div
      className="calendar"
    >
      <Connect(Navigation)>
        <Navigation
          goToToday={[Function]}
          month={2020-08-31T12:12:19.313Z}
          nextMonth={[Function]}
          prevMonth={[Function]}
        >
          <div
            className="navigation"
          >
            <Button
              clickHandler={[Function]}
              filled={false}
            >
              <div
                className="button"
                onClick={[Function]}
              >
                <ForwardRef
                  color="currentColor"
                  size="1em"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
                      fillRule="evenodd"
                    />
                  </svg>
                </ForwardRef>
              </div>
            </Button>
            <Button
              clickHandler={[Function]}
              filled={true}
            >
              <div
                className="button filled"
                onClick={[Function]}
              >
                Сегодня
              </div>
            </Button>
            <Button
              clickHandler={[Function]}
              filled={false}
            >
              <div
                className="button"
                onClick={[Function]}
              >
                <ForwardRef
                  color="currentColor"
                  size="1em"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
                      fillRule="evenodd"
                    />
                  </svg>
                </ForwardRef>
              </div>
            </Button>
          </div>
        </Navigation>
      </Connect(Navigation)>
      <Connect(MonthLabel)>
        <MonthLabel
          dispatch={[Function]}
          month={2020-08-31T12:12:19.313Z}
        >
          <div
            className="month-label"
          >
            август 2020
          </div>
        </MonthLabel>
      </Connect(MonthLabel)>
      <WeekdayLabel>
        <div
          className="week label"
        >
          <div
            className="day label"
            key="ПН"
          >
            ПН
          </div>
          <div
            className="day label"
            key="ВТ"
          >
            ВТ
          </div>
          <div
            className="day label"
            key="СР"
          >
            СР
          </div>
          <div
            className="day label"
            key="ЧТ"
          >
            ЧТ
          </div>
          <div
            className="day label"
            key="ПТ"
          >
            ПТ
          </div>
          <div
            className="day label"
            key="СБ"
          >
            СБ
          </div>
          <div
            className="day label"
            key="ВС"
          >
            ВС
          </div>
        </div>
      </WeekdayLabel>
      <Connect(Month)>
        <Month
          date={2020-08-31T12:12:19.313Z}
          dispatch={[Function]}
        >
          <div
            className="month"
          >
            <Week
              days={
                Array [
                  2020-07-31T21:00:00.000Z,
                  2020-08-01T21:00:00.000Z,
                ]
              }
              key="1596229200000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-07-31T21:00:00.000Z}
                  key="1596229200000"
                >
                  <Day
                    date={2020-07-31T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      1
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-01T21:00:00.000Z}
                  key="1596315600000"
                >
                  <Day
                    date={2020-08-01T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      2
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
            <Week
              days={
                Array [
                  2020-08-02T21:00:00.000Z,
                  2020-08-03T21:00:00.000Z,
                  2020-08-04T21:00:00.000Z,
                  2020-08-05T21:00:00.000Z,
                  2020-08-06T21:00:00.000Z,
                  2020-08-07T21:00:00.000Z,
                  2020-08-08T21:00:00.000Z,
                ]
              }
              key="1596402000000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-08-02T21:00:00.000Z}
                  key="1596402000000"
                >
                  <Day
                    date={2020-08-02T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      3
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-03T21:00:00.000Z}
                  key="1596488400000"
                >
                  <Day
                    date={2020-08-03T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      4
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-04T21:00:00.000Z}
                  key="1596574800000"
                >
                  <Day
                    date={2020-08-04T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      5
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-05T21:00:00.000Z}
                  key="1596661200000"
                >
                  <Day
                    date={2020-08-05T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      6
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-06T21:00:00.000Z}
                  key="1596747600000"
                >
                  <Day
                    date={2020-08-06T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      7
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-07T21:00:00.000Z}
                  key="1596834000000"
                >
                  <Day
                    date={2020-08-07T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      8
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-08T21:00:00.000Z}
                  key="1596920400000"
                >
                  <Day
                    date={2020-08-08T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      9
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
            <Week
              days={
                Array [
                  2020-08-09T21:00:00.000Z,
                  2020-08-10T21:00:00.000Z,
                  2020-08-11T21:00:00.000Z,
                  2020-08-12T21:00:00.000Z,
                  2020-08-13T21:00:00.000Z,
                  2020-08-14T21:00:00.000Z,
                  2020-08-15T21:00:00.000Z,
                ]
              }
              key="1597006800000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-08-09T21:00:00.000Z}
                  key="1597006800000"
                >
                  <Day
                    date={2020-08-09T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      10
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-10T21:00:00.000Z}
                  key="1597093200000"
                >
                  <Day
                    date={2020-08-10T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      11
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-11T21:00:00.000Z}
                  key="1597179600000"
                >
                  <Day
                    date={2020-08-11T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      12
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-12T21:00:00.000Z}
                  key="1597266000000"
                >
                  <Day
                    date={2020-08-12T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      13
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-13T21:00:00.000Z}
                  key="1597352400000"
                >
                  <Day
                    date={2020-08-13T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      14
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-14T21:00:00.000Z}
                  key="1597438800000"
                >
                  <Day
                    date={2020-08-14T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      15
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-15T21:00:00.000Z}
                  key="1597525200000"
                >
                  <Day
                    date={2020-08-15T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      16
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
            <Week
              days={
                Array [
                  2020-08-16T21:00:00.000Z,
                  2020-08-17T21:00:00.000Z,
                  2020-08-18T21:00:00.000Z,
                  2020-08-19T21:00:00.000Z,
                  2020-08-20T21:00:00.000Z,
                  2020-08-21T21:00:00.000Z,
                  2020-08-22T21:00:00.000Z,
                ]
              }
              key="1597611600000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-08-16T21:00:00.000Z}
                  key="1597611600000"
                >
                  <Day
                    date={2020-08-16T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      17
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-17T21:00:00.000Z}
                  key="1597698000000"
                >
                  <Day
                    date={2020-08-17T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      18
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-18T21:00:00.000Z}
                  key="1597784400000"
                >
                  <Day
                    date={2020-08-18T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      19
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-19T21:00:00.000Z}
                  key="1597870800000"
                >
                  <Day
                    date={2020-08-19T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      20
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-20T21:00:00.000Z}
                  key="1597957200000"
                >
                  <Day
                    date={2020-08-20T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      21
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-21T21:00:00.000Z}
                  key="1598043600000"
                >
                  <Day
                    date={2020-08-21T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      22
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-22T21:00:00.000Z}
                  key="1598130000000"
                >
                  <Day
                    date={2020-08-22T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      23
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
            <Week
              days={
                Array [
                  2020-08-23T21:00:00.000Z,
                  2020-08-24T21:00:00.000Z,
                  2020-08-25T21:00:00.000Z,
                  2020-08-26T21:00:00.000Z,
                  2020-08-27T21:00:00.000Z,
                  2020-08-28T21:00:00.000Z,
                  2020-08-29T21:00:00.000Z,
                ]
              }
              key="1598216400000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-08-23T21:00:00.000Z}
                  key="1598216400000"
                >
                  <Day
                    date={2020-08-23T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      24
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-24T21:00:00.000Z}
                  key="1598302800000"
                >
                  <Day
                    date={2020-08-24T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      25
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-25T21:00:00.000Z}
                  key="1598389200000"
                >
                  <Day
                    date={2020-08-25T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      26
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-26T21:00:00.000Z}
                  key="1598475600000"
                >
                  <Day
                    date={2020-08-26T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      27
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-27T21:00:00.000Z}
                  key="1598562000000"
                >
                  <Day
                    date={2020-08-27T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day"
                      onClick={[Function]}
                    >
                      28
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-28T21:00:00.000Z}
                  key="1598648400000"
                >
                  <Day
                    date={2020-08-28T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      29
                    </div>
                  </Day>
                </Connect(Day)>
                <Connect(Day)
                  date={2020-08-29T21:00:00.000Z}
                  key="1598734800000"
                >
                  <Day
                    date={2020-08-29T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day weekend"
                      onClick={[Function]}
                    >
                      30
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
            <Week
              days={
                Array [
                  2020-08-30T21:00:00.000Z,
                ]
              }
              key="1598821200000"
            >
              <div
                className="week"
              >
                <Connect(Day)
                  date={2020-08-30T21:00:00.000Z}
                  key="1598821200000"
                >
                  <Day
                    date={2020-08-30T21:00:00.000Z}
                    now={2020-08-31T12:12:19.313Z}
                    selectDate={[Function]}
                    selected={2020-08-31T12:12:19.313Z}
                    tasksMap={Object {}}
                  >
                    <div
                      className="day selected today"
                      onClick={[Function]}
                    >
                      31
                    </div>
                  </Day>
                </Connect(Day)>
              </div>
            </Week>
          </div>
        </Month>
      </Connect(Month)>
    </div>
  </Calendar>
</Provider>
`;
